projects/core/src/checkout/events/checkout-event.builder.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import { ofType } from '@ngrx/effects';
+import { ActionsSubject } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { EventService } from '../../event/event.service';
+import { ActionToEventMapping } from '../../state/event/action-to-event-mapping';
+import { createFrom } from '../../util/create-from';
+import { CheckoutActions } from '../store/actions/index';
+import { OrderPlaced } from './checkout.events';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CheckoutEventBuilder {
+  constructor(
+    protected actionsSubject: ActionsSubject,
+    protected eventService: EventService
+  ) {
+    this.register();
+  }
+
+  /**
+   * Registers checkout events
+   */
+  protected register(): void {
+    this.orderPlacedEvent();
+  }
+
+  /**
+   * Register an order successfully placed event
+   */
+  protected orderPlacedEvent(): void {
+    this.registerMapped({
+      action: CheckoutActions.PLACE_ORDER_SUCCESS,
+      event: OrderPlaced,
+    });
+  }
+
+  /**
+   * Registers a stream of target events mapped from the source actions.
```

Comment:
Do we need this level of abstraction here? What about using `StateEventService`?

Reply:
It was modelled based on `projects/core/src/cart/event/cart-event.builder.ts`, but I can look into the StateEventService